### PR TITLE
fix game instance

### LINKED ISSUE #83: [soa_test_debugging] Fix game instance issue
issue: run SOA_test.py will return an error 

`AttributeError: 'Jsonable' object has no attribute 'victory'`

due to `get_best_proposer(..., game)` in `utils.py `

The current logic is to copy a parent class of input of `game` --- we assume that game input is `NetworkGame` for TACC https. However, this will be an error when we input a `Game` instance. 

`Game` https://github.com/SHADE-AI/diplomacy/blob/master/diplomacy/engine/game.py
`NetworkGame` (`Game` as parent) https://github.com/SHADE-AI/diplomacy/blob/master/diplomacy/client/network_game.py

### PR DESCRIPTION
**Pull request by wwongkamjan**
resolve #83

### CODE REVIEW

File: tests/bot_tests/SOA_test.py [hunk 1]
```diff
@@ -26,11 +26,11 @@
 
 class TestSOABot():
     def test(self):
-        # start_io_loop(self.test_play)
+        start_io_loop(self.test_play)
         # start_io_loop(self.test_stance)
```
Comment (trigaten) [inline tests/bot_tests/SOA_test.py:30]: remove commented out code

File: src/baseline_bots/utils.py [hunk 2]
```diff
@@ -414,8 +414,13 @@ def get_best_orders(bot, proposal_order: dict, shared_order: dict):
     
     def __deepcopy__(game):
```
Comment (trigaten) [inline src/baseline_bots/utils.py:415]: It would be nice to see a test for this function

File: tests/bot_tests/SOA_test.py (hunk 1 above)
Comment (wwongkamjan) [reply tests/bot_tests/SOA_test.py:30]: we can run only one test at a time due to ioloop. So those test functions were there even with comments, just to later uncomment and run test one by one.

File: src/baseline_bots/utils.py (hunk 2 above)
Comment (wwongkamjan) [reply src/baseline_bots/utils.py:415]: this function is directly copied from the main diplomacy. I did not do much of any update  just checking stance

File: tests/bot_tests/SOA_test.py (hunk 1 above)
Comment (trigaten) [reply tests/bot_tests/SOA_test.py:30]: There is certainly a way around that. What have you tried/had difficulty with?

**Review by kartik2112 — APPROVED** [review]

File: src/baseline_bots/bots/smart_order_accepter_bot.py [hunk 3]
```diff
@@ -344,7 +345,7 @@ def __call__(self, rcvd_messages: List[Tuple[int, Message]]):
         # if there is any better proposal orders that has a state value more than ours, then do it. If not, just follow the base orders.
         valid_proposal_orders[self.power_name] = orders
 
-        # best_proposer, best_orders = yield from get_best_orders(self, valid_proposal_orders, shared_orders)
+        best_proposer, best_orders = yield from get_best_orders(self, valid_proposal_orders, shared_orders)
         best_orders, best_proposer = orders, list(self.alliances.keys())[0] if self.alliances else ""
         # print("debug: Fetched best orders")
```
Comment (trigaten) [inline src/baseline_bots/bots/smart_order_accepter_bot.py:350]: remove prints

File: tests/bot_tests/SOA_test.py (hunk 1 above)
Comment (wwongkamjan) [reply tests/bot_tests/SOA_test.py:30]: I have tried `startioloop(test)` but it doesn't work. but this is not a focus for this branch. The `SOA_test.py` file is already in the main right now.

File: src/baseline_bots/bots/smart_order_accepter_bot.py (hunk 3 above)
Comment (wwongkamjan) [reply src/baseline_bots/bots/smart_order_accepter_bot.py:350]: this is also not the focus in this branch. I think it is removed in the main?

File: tests/bot_tests/SOA_test.py [hunk 4]
```diff
@@ -227,7 +227,7 @@ def test_get_best_orders(self):
                 
                 state_value[power_name] = yield get_state_value(soa_bot, sim_game, soa_bot.power_name)
             
-            print(state_value)
+            print('state value from power proposals',state_value)
```
[the remark this answers is not in the document]
Comment (wwongkamjan) [reply tests/bot_tests/SOA_test.py:230]: No, state values are not certain values. They depends on dipnet orders, and there's prob that there won't get the same output though we input the same state. This is to check if state value and best proposer are related

**Pull request merged**